test/minter-suite-minters/MinterMerkle/MinterMerkleV4.test.ts

```diff
@@ -0,0 +1,725 @@
+
+      // populate Merkle elements for projects zero, one, and two
+      const elementsProjectZero = [];
+      const elementsProjectOne = [];
+      const elementsProjectTwo = [];
+
+      elementsProjectZero.push(
+        this.accounts.deployer.address,
+        this.accounts.artist.address,
+        this.accounts.additional.address,
+        this.accounts.user.address,
+        this.accounts.user2.address
+      );
+      elementsProjectOne.push(
+        this.accounts.user.address,
+        this.accounts.additional2.address
+      );
+      elementsProjectTwo.push(this.accounts.additional.address);
+
+      // build Merkle trees for projects zero, one, and two
+      this.merkleTreeZero = new MerkleTree(
+        elementsProjectZero.map((_addr) => hashAddress(_addr)),
+        keccak256,
+        {
+          sortPairs: true,
+        }
+      );
+      this.merkleTreeOne = new MerkleTree(
+        elementsProjectOne.map((_addr) => hashAddress(_addr)),
+        keccak256,
+        {
+          sortPairs: true,
+        }
+      );
+      this.merkleTreeTwo = new MerkleTree(
+        elementsProjectTwo.map((_addr) => hashAddress(_addr)),
+        keccak256,
+        {
+          sortPairs: true,
+        }
+      );
+
+      // update Merkle root for projects zero and one on minter
+      const merkleRootZero = this.merkleTreeZero.getHexRoot();
+      const merkleRootOne = this.merkleTreeOne.getHexRoot();
+      // Merkle root two intentionally not set
+      await this.minter
+        .connect(this.accounts.artist)
+        .updateMerkleRoot(this.projectZero, merkleRootZero);
+      await this.minter
+        .connect(this.accounts.artist)
+        .updateMerkleRoot(this.projectOne, merkleRootOne);
+
+      // mock ERC20 genArt721Core
+      const ERC20Factory = await ethers.getContractFactory("ERC20Mock");
+      this.ERC20Mock = await ERC20Factory.deploy(
+        ethers.utils.parseEther("100")
+      );
+    });
+
+    describe("common MinterMerkle tests", async () => {
+      await MinterMerkle_Common();
+    });
+
+    // describe("constructor", async function () {
+    //   it("emits an event indicating dependency registry in constructor", async function () {
+    //     const contractFactory = await ethers.getContractFactory(
+    //       this.targetMinterName
+    //     );
+    //     const tx = await contractFactory.deploy(
+    //       this.genArt721Core.address,
+    //       this.minterFilter.address,
+    //       this.delegationRegistry.address
+    //     );
+    //     const receipt = await tx.deployTransaction.wait();
+    //     // target event is the last log
+    //     const targetLog = receipt.logs[0];
+    //     // expect "PlatformUpdated" event as log 0
+    //     await expect(targetLog.topics[0]).to.be.equal(
+    //       ethers.utils.keccak256(
+    //         ethers.utils.toUtf8Bytes("DelegationRegistryUpdated(address)")
+    //       )
+    //     );
```

Comment:
@lyaunzbe I'm confused why we have this long set of commented out tests – also, I do wonder if we should just be updating our testing suite to only test the latest of each minter version, rather than having additional tests for each version.. WDYT here @ryley-o ?

Reply:
ah thank you for catching this @jakerockland this was a relic of me debugging something (gas values from common tests were causing failures in circle), but this should be uncommented!

regarding your suggestion for overall testing strategy, I don't have too strong of an opinion here. Would agree that only  testing latest minter versions would be streamlined/nice dev ex generally, but also in the context of smart contract dev specifically and the fact that earlier versions will continue to be used in the wild (something we cant really control), it may be nice to have those older tests for reference more than anything. curious to hear more of your thoughts as well @ryley-o

Reply:
I've added an issue regarding archiving tests associated with deprecated minters here: https://github.com/ArtBlocks/artblocks-contracts/issues/420

For new minters, I think utilizing the common tests to achieve 100% coverage is good practice. Hopefully the common tests make the lift relatively minimal when small additional functionality is added in PRs like these. Happy to revisit if we think it is overkill for this kind of PR though!